sdk/openai/Azure.AI.OpenAI/src/Custom/StreamingChatChoice.cs

```diff
@@ -33,7 +33,8 @@ public class StreamingChatChoice
         /// Normal termination typically provides "stop" and encountering token limits in a request typically
         /// provides "length." If no value is present, this StreamingChoice is still in progress.
         /// </remarks>
-        public string FinishReason => GetLocked(() => _baseChoices.Last().FinishReason);
+        public CompletionsFinishReason? FinishReason => GetLocked(() =>
+            _baseChoices.Last().FinishReason == null ? null : _baseChoices.Last().FinishReason);
```

Comment:
Versus using the `GetLocked` synchronization helper? I wasn't sure what would end up being concurrency-safe via LINQ since `_baseChoices` can be updated while it's being read here, though intuitively I would hope `.Last()` isn't problematic for a purely additive case.

Do you see any issues with keeping all the concurrent collection use under the lock just to keep it consistent, though?

Reply:
No, I was just referring to reducing this:
```csharp
_baseChoices.Last().FinishReason == null ? null : _baseChoices.Last().FinishReason
```

To this:
```
_baseChoices.Last().FinishReason
```

Like it is in StreamingChoice.cs. In other words, you don't need to check for null.